### ISSUE TITLE
sendTags not working without permission and in background mode

### ISSUE THREAD

**Issue by pcki11** [OneSignal/OneSignal-iOS-SDK#331]
I am trying to send client tags immediately after SDK initialization, but _before_ the user has granted the permission to use push notifications and I have noticed an unstable behaviour:
1. If using [OneSignal sendTags:pushTags] the tags are sometimes being sent, and sometimes not.
2. If using [OneSignal sendTags:pushTags onSuccess:^(NSDictionary* result){} onFailure:^(NSError* error){}] the tags are never sent.

My second issue is using [OneSignal sendTags:pushTags] in background mode, but I have found a workaround:
```
__block UIBackgroundTaskIdentifier backgroundTaskId = [[UIApplication sharedApplication] beginBackgroundTaskWithExpirationHandler:^{
    backgroundTaskId = UIBackgroundTaskInvalid;
}];
[OneSignal sendTags:pushTags onSuccess:^(NSDictionary *result) {
    [[UIApplication sharedApplication] endBackgroundTask:backgroundTaskId];
} onFailure:^(NSError *error) {
    [[UIApplication sharedApplication] endBackgroundTask:backgroundTaskId];
}];
```
Unfortunately, this method breaks if the user has not yet granted push notifications permission.

**Comment by Nightsd01** [comment]
@pcki11 Great catch, thank you for reporting this. We have identified this as a bug with the SDK and will be fixing it shortly.

**Comment by Nightsd01** [comment]
By the way, I want to make it clear, the tags _are_ being sent, it's just that they are sent with a delay and the callbacks don't get called.

**Comment by Nightsd01** [comment]
Fixed in #332

**Comment by Nightsd01** [comment]
PR #332 is now merged, thanks for reporting this issue @pcki11
